### PR TITLE
Interval dates added

### PR DESCRIPTION
**Pull request by CourtneyGosselin**
## Ticket(s)
[DIG-1415](https://candig.atlassian.net/browse/DIG-1415)

## Description
Update all fields to work with interval dates based on decisions in data-portal channel. All current pages (summary, search, patient info) have had changes to work with itnerval dates.

## Screenshots
image
image
image
image

## Types of Change(s)

-   [ ] 🪲 Bug fix (non-breaking change that fixes an issue)
-   [x] ✨ New feature (non-breaking change that adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:

-   [x] My change requires a change to the documentation
-   [ ] I have updated the documentation accordingly
-   [ ] Prettier linter doesn't return errors
-   [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc.
-   [x] Locally tested
-   [ ] Dev server tested
-   [ ] Production tested when merging into stable/production branch
-   [ ] [Runbook tasks](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) pass locally/on UHN-Dev
-   [ ] If visuals have changed, [Runbook](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) has been updated with new screenshots

[DIG-1415]: https://candig.atlassian.net/browse/DIG-1415?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/views/clinicalGenomic/useClinicalPatientData.js [hunk 1]
```diff
@@ -27,10 +27,10 @@ function useClinicalPatientData(patientId, programId) {
                     value !== null &&
                     !(
                         (Array.isArray(value) && value.length === 0) || // Exclude empty arrays
-                        typeof value === 'object' || // Exclude all objects
                         value === '' ||
                         key === ''
-                    )
+                    ) &&
+                    (!(typeof obj[key] === 'object') || (typeof obj[key] === 'object' && 'month_interval' in obj[key]))
```
Comment (OrdiNeu) [inline src/views/clinicalGenomic/useClinicalPatientData.js]: Might want a comment here explaining that this is to accept intervals

File: src/views/clinicalGenomic/widgets/clinicalData.js [hunk 2]
```diff
@@ -33,6 +33,14 @@ function ClinicalView() {
                     // Make sure each row has an ID and a deceased status
                     patient.id = index;
                     patient.deceased = !!patient.date_of_death;
+                    if (patient.date_of_birth && patient.date_of_death) {
+                        const ageInMonths = patient.date_of_death.month_interval - patient.date_of_birth.month_interval;
+                        patient.date_of_death = Math.floor(ageInMonths / 12);
+                        patient.date_of_birth = Math.floor(-patient.date_of_birth.month_interval / 12);
+                    } else {
+                        delete patient.date_of_birth;
```
Comment (OrdiNeu) [inline src/views/clinicalGenomic/widgets/clinicalData.js:43]: This removes DOB if they haven't died yet -- is this intentional? If so, we'll want a comment with the reason.

**Review by OrdiNeu — APPROVED** [review]
LGTM :heavy_check_mark:

File: src/views/clinicalGenomic/widgets/clinicalData.js (hunk 2 above)
Comment (CourtneyGosselin) [reply src/views/clinicalGenomic/widgets/clinicalData.js:43]: Unintentional missed an if else

**Pull request merged**